cmd/protecodeExecuteScan.go

```diff
@@ -0,0 +1,317 @@
+	}
+
+	return image
+}
+
+func getDockerImage(scanImage string, registryURL string, path string, includeLayers bool) (string, string) {
+	fileName := scanImage
+
+	cachePath := "./cache"
+	cacheProtecodeImagePath := "/protecodeImage"
+	cacheImagePath := filepath.Join(cachePath, cacheProtecodeImagePath)
+	os.Mkdir(cacheImagePath, 600)
+
+	image := getImage(scanImage, registryURL, includeLayers, cacheImagePath)
+
+	tarFileName := filepath.Join(cacheImagePath, fileName)
+
+	if !pkgutil.IsTar(scanImage) {
+
+		fileName = fmt.Sprintf("%v.tar", strings.ReplaceAll(scanImage, "/", "_"))
+		tarFileName = filepath.Join(cachePath, fileName)
+
+		tarImageData(tarFileName, image, cacheImagePath)
+	}
+
+	var resultFilePath string
+
+	if len(path) <= 0 {
+		resultFilePath = filepath.Join(cachePath, fileName)
+		if len(resultFilePath) <= 0 {
+			log.Entry().Fatalf("Protecode scan failed, there is no file path configured  : %v (filename:%v, PSPath: %v)", path, fileName, image.FSPath)
+		}
+	}
+
+	return fileName, resultFilePath
+}
+
+func tarImageData(tarFileName string, image pkgutil.Image, cacheImagePath string) {
+	tarFile, err := os.Create(tarFileName)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Protecode scan failed, error during create tar for the docker image")
+	}
+	if err := os.Chmod(tarFileName, 0644); err != nil {
+		log.Entry().WithError(err).Fatal("Protecode scan failed, error during create tar for the docker image")
+	}
+	defer tarFile.Close()
+
+	reference, err := name.ParseReference(image.Digest.String(), name.WeakValidation)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Protecode scan failed, not possible to parse reference of docker image")
+	}
+	err = tarball.Write(reference, image.Image, tarFile)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Protecode scan failed, error during create tar archive of docker image via tarball")
+	}
+
+	os.RemoveAll(cacheImagePath)
+}
+
+func getURLAndFileNameFromDockerImage(scanImage string, registryURL string) string {
+
+	completeURL := scanImage
+
+	if len(registryURL) > 0 {
+		if strings.HasSuffix(registryURL, "/") {
+			completeURL = fmt.Sprintf("remote://%v%v", registryURL, scanImage)
+		} else {
+			completeURL = fmt.Sprintf("remote://%v/%v", registryURL, scanImage)
+		}
+	}
+
+	if len(completeURL) <= 0 {
+		log.Entry().Fatal("Protecode scan failed, there is no scan image configured")
+	}
+
+	return completeURL
+}
+
+func executeProtecodeScan(client protecode.Protecode, config *protecodeExecuteScanOptions, fileName string, writeReportToFile func(resp io.ReadCloser, reportFileName string) error) (map[string]int, int) {
+
+	var parsedResult map[string]int = make(map[string]int)
+	//load existing product by filename
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, load existing product")
+	}
+	productID := client.LoadExistingProduct(config.ProtecodeGroup, config.ReuseExisting)
+
+	// check if no existing is found or reuse existing is false
```

Comment:
I don't understand the comment. Maybe you can make it a little bit more specific?

Reply:
That was part of a contribution 🤣. But actually:  "Based on the value of 'reuseExisting' check if we either got an existing product from the backend or continue" would be more speaking than the existing one.